docs/docs/classic-ml/deep-learning/tensorflow/guide/index.mdx

```diff
@@ -0,0 +1,676 @@
+
+  1. **Format Conversion**: The model is converted to a generic MLflow `pyfunc` model
+  2. **Metadata Creation**: Model metadata is stored, including dependencies and signature
+  3. **Artifact Storage**: The model is saved to the MLflow artifact store
+  4. **Loading Capability**: The model can be loaded back as either a native TensorFlow model or a generic `pyfunc` model
+
+  This approach enables consistent model management regardless of the framework used.
+</details>
+
+### Model Formats
+
+By default, MLflow saves TensorFlow models in the TensorFlow SavedModel format (compiled graph), which is ideal for deployment. You can also save in other formats:
+
+```python
+# Save in H5 format (weights only)
+mlflow.tensorflow.log_model(
+    model, name="model", keras_model_kwargs={"save_format": "h5"}
+)
+
+# Save in native Keras format
+mlflow.tensorflow.log_model(
+    model, name="model", keras_model_kwargs={"save_format": "keras"}
+)
+```
+
+<details>
+  <summary>Comparing Model Formats</summary>
+
+  #### TensorFlow SavedModel (Default)
+  - ✅ **Complete Serialization**: Includes model architecture, weights, and compilation information
+  - ✅ **Deployment Ready**: Optimized for production environments
+  - ✅ **TensorFlow Serving**: Compatible with TensorFlow Serving
+  - ✅ **Cross-Platform**: Can be deployed across different platforms
+
+  #### H5 Format
+  - ✅ **Weights Storage**: Efficiently stores model weights
+  - ✅ **Smaller Size**: Generally smaller than SavedModel format
+  - ❌ **Limited Information**: Doesn't include the full computation graph
+  - ❌ **Deployment Limitations**: Not ideal for certain deployment scenarios
+
+  #### Keras Format
+  - ✅ **Native Keras**: Uses Keras' native serialization
+  - ✅ **Compatibility**: Works well with newer Keras versions
+  - ❌ **Deployment**: May require additional steps for deployment
+
+  For most production use cases, the default SavedModel format is recommended. For more details, see [TensorFlow Save and Load Guide](https://www.tensorflow.org/guide/keras/save_and_serialize).
+</details>
+
+### Model Signatures
+
+A model signature describes the expected input and output formats of your model. While optional, it's a best practice for better model understanding and validation:
+
+```python
+import mlflow
+import tensorflow as tf
+import numpy as np
+from tensorflow import keras
+from mlflow.types import Schema, TensorSpec
+from mlflow.models import ModelSignature
+
+# Define model
+model = keras.Sequential(
+    [
+        keras.Input([28, 28, 3]),
+        keras.layers.Conv2D(8, 2),
+        keras.layers.MaxPool2D(2),
+        keras.layers.Flatten(),
+        keras.layers.Dense(2),
+        keras.layers.Softmax(),
+    ]
+)
+
+# Define input schema
+input_schema = Schema(
+    [
+        TensorSpec(np.dtype(np.float32), (-1, 28, 28, 3), "input"),
+    ]
+)
+
+# Create signature with input schema
+signature = ModelSignature(inputs=input_schema)
```

Comment:
Same here, I would prefer to put infer_signature example here and this manual construction inside the details